src/__tests__/message-id.integration.test.js

```diff
@@ -0,0 +1,273 @@
+      fragmentMessageIds: fragmentMessageIds,
+      conversationIdFromEhrIn: conversationIdFromEhrIn
+    };
+
+    const ehrRepoScope = nock(ehrRepoUrl, ehrRepoHeaders)
+      .get(`/patients/${nhsNumber}`)
+      .reply(200, ehrRepoCoreResponse);
+    return ehrRepoScope;
+  }
+  // ================= END HELPER FUNCTIONS =================
+
+  describe('EHR core uses new message IDs', () => {
+    it('should update the message IDs of the EHR core and referenced fragments', async () => {
+      // given
+      const ehrCore = readFileSync('src/__tests__/data/ehr_with_fragments/ehr-core', 'utf8');
+
+      const ebXML = JSON.parse(ehrCore).ebXML;
+      const oldMessageId = await extractMessageId(ebXML);
+      const oldReferencedFragmentIds = await extractReferencedFragmentMessageIds(ebXML);
+
+      // set up mocks
+      const ehrRepoScope = setUpMockForEhrRepoCoreMessage();
+      const s3Scope = nock(ehrCorePresignedUrl).get('/').reply(200, ehrCore);
+      const gp2gpMessengerGetODSScope = setUpMockForGp2gpGetOdsCode();
+
+      let gp2gpMessengerPostBody;
+      const storePostBody = body => {
+        gp2gpMessengerPostBody = body;
+        return true;
+      };
+
+      const gp2gpMessengerSendCoreScope = nock(gp2gpUrl, gp2gpHeaders)
+        .post('/ehr-out-transfers/core', body => storePostBody(body))
+        .reply(204);
+
+      // when
+      await transferOutEhrCore({ conversationId, nhsNumber, odsCode, ehrRequestId });
+
+      // then
+      expect(ehrRepoScope.isDone()).toBe(true);
+      expect(gp2gpMessengerGetODSScope.isDone()).toBe(true);
+      expect(s3Scope.isDone()).toBe(true);
+      expect(gp2gpMessengerSendCoreScope.isDone()).toBe(true);
+
+      expect(gp2gpMessengerPostBody).toEqual({
+        conversationId: conversationId,
+        odsCode: odsCode,
+        ehrRequestId: ehrRequestId,
+        coreEhr: expect.anything()
+      });
+
+      const outBoundEhrCore = gp2gpMessengerPostBody.coreEhr;
+
+      const outBoundEhrCoreAsString = JSON.stringify(outBoundEhrCore);
+
+      expect(outBoundEhrCoreAsString.includes(oldMessageId)).toBe(false);
+      for (let oldFragmentId of oldReferencedFragmentIds) {
+        expect(outBoundEhrCoreAsString.includes(oldFragmentId)).toBe(false);
+      }
+    });
+  });
+
+  describe('Message fragments use new message IDs', () => {
+    /*
+      The structure of the files within the ehr_with_fragments directory is:
+
+      EHR-core
+      | fragment-1
+      | fragment-2
+      | | fragment-2-1
+      | | fragment-2-2
+
+     */
+
+    function makePresignedUrlForFragment(messageId) {
+      return `http://fake-fragment-presign-url/${messageId}`;
+    }
+
+    function setUpMockForEhrRepoFragmentMessage(messageId) {
+      const ehrFragmentPresignedUrl = makePresignedUrlForFragment(messageId);
+
+      return (
+        nock(ehrRepoUrl, ehrRepoHeaders)
+          // TODO: currently this endpoint of ehr-repo doesn't work as expect. we might need to change this endpoint later
```

Comment:
Has this been addressed? If not shall this be a raised as a new ticket to investigate and address?